influxdb3_write/src/write_buffer/flusher.rs

```diff
@@ -0,0 +1,245 @@
+
+// buffered writes should only fail if the underlying WAL throws an error. They are validated before they
+// are buffered. If there is an error, it'll be here
+#[derive(Debug, Clone)]
+pub enum BufferedWriteResult {
+    Success(WriteSummary),
+    Error(String),
+}
+
+#[derive(Debug)]
+pub struct WriteBufferFlusher {
+    join_handle: Mutex<Option<tokio::task::JoinHandle<()>>>,
+    wal_io_handle: Mutex<Option<std::thread::JoinHandle<()>>>,
+    #[allow(dead_code)]
+    shutdown_tx: watch::Sender<()>,
+    buffer_tx: mpsc::Sender<BufferedWrite>,
+}
+
+impl WriteBufferFlusher {
+    pub fn new(segment_state: Arc<RwLock<SegmentState>>) -> Self {
+        let (shutdown_tx, shutdown_rx) = watch::channel(());
+        let (buffer_tx, buffer_rx) = mpsc::channel(BUFFER_CHANNEL_LIMIT);
+        let (io_flush_tx, io_flush_rx) = std::sync::mpsc::channel();
+        let (io_flush_notify_tx, io_flush_notify_rx) = std::sync::mpsc::channel();
+
+        let flusher = Self {
+            join_handle: Default::default(),
+            wal_io_handle: Default::default(),
+            shutdown_tx,
+            buffer_tx,
+        };
+
+        let wal_op_buffer_segment_state = Arc::clone(&segment_state);
+
+        *flusher.wal_io_handle.lock() = Some(
+            std::thread::Builder::new()
+                .name("write buffer io flusher".to_string())
+                .spawn(move || {
+                    run_io_flush(segment_state, io_flush_rx, io_flush_notify_tx);
+                })
+                .expect("failed to spawn write buffer io flusher thread"),
+        );
+
+        *flusher.join_handle.lock() = Some(tokio::task::spawn(async move {
+            run_wal_op_buffer(
+                wal_op_buffer_segment_state,
+                buffer_rx,
+                io_flush_tx,
+                io_flush_notify_rx,
+                shutdown_rx,
+            )
+            .await;
+        }));
+
+        flusher
+    }
+
+    pub async fn write_to_open_segment(
+        &self,
+        db_name: NamespaceName<'static>,
+        table_batches: HashMap<String, TableBatch>,
+        wal_op: WalOp,
+    ) -> crate::write_buffer::Result<WriteSummary> {
+        let (response_tx, response_rx) = oneshot::channel();
+
+        self.buffer_tx
+            .send(BufferedWrite {
+                wal_op,
+                database_write: DatabaseWrite::new(db_name, table_batches),
+                response_tx,
+            })
+            .await
+            .expect("wal op buffer thread is dead");
+
+        let summary = response_rx.await.expect("wal op buffer thread is dead");
+
+        match summary {
+            BufferedWriteResult::Success(summary) => Ok(summary),
+            BufferedWriteResult::Error(e) => Err(Error::BufferSegmentError(e)),
+        }
+    }
+}
+
+async fn run_wal_op_buffer(
+    segment_state: Arc<RwLock<SegmentState>>,
+    mut buffer_rx: mpsc::Receiver<BufferedWrite>,
+    io_flush_tx: std::sync::mpsc::Sender<Vec<WalOp>>,
```

Comment:
I think we can do an import at the top like:

```rust
use std::sync::mpsc::Sender as StdSender;
use std::sync::mpsc::Receiver as StdReceiver;
```

letting you just write `StdSender<Vec<WalOp>>` and reducing the line noise.